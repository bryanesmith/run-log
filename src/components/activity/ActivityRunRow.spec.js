import React from 'react';
import ActivityRunRow from './ActivityRunRow.jsx';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

const Samples = {
  minimal: {
    "@id": "_:n1",
    "@type": "Run"
  },
  sample1: {
    "@id": "_:n1",
    "@type": "Run",
    "date": "2017-09-10",
    "category": "casual",
    "distance": 5.00,
    "duration": "PT50M00S",
    "notes": "Just a test."
  }
};

//
// To update snaphot: $ jest --updateSnapshot
//
test('Sample event with all fields defined should match snapshot', () => {

  const component = renderer.create(
    <ActivityRunRow event={Samples.sample1} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Minimal event should match snapshot.', () => {

  const component = renderer.create(
    <ActivityRunRow event={Samples.minimal} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Minimal event should contain expected default values', () => {

  const row = shallow(
    <ActivityRunRow event={Samples.minimal} />
  );

  const classes = ['data-date', 'data-category', 'data-distance', 'data-duration', 'data-notes'];

  classes.forEach(className => {
    const ele = row.find(`.${className} > .value`);
    expect(ele.length).toBe(1);
    expect(ele.text()).toBe('-');
  });
});
