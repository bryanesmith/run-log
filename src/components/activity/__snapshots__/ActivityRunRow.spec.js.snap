// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Minimal event should match snapshot. 1`] = `
<tr
  className="activity-run"
>
  <td
    className="data-date"
  >
    <span
      className="value"
    >
      -
    </span>
  </td>
  <td
    className="data-category"
  >
    <span
      className="glyphicon glyphicon-road"
    />
     
    <span
      className="value"
    >
      -
    </span>
  </td>
  <td
    className="data-distance"
  >
    <span
      className="value"
    >
      -
    </span>
  </td>
  <td
    className="data-duration"
  >
    <span
      className="value"
    >
      -
    </span>
  </td>
  <td
    className="data-notes"
  >
    <span
      className="value"
    >
      -
    </span>
  </td>
  <td
    className="data-actions"
  >
    <a
      className="glyphicon glyphicon-heart-empty"
      href="#"
    />
    <a
      className="glyphicon glyphicon-pencil"
      href="#"
    />
    <a
      className="glyphicon glyphicon glyphicon-trash"
      href="#"
    />
  </td>
</tr>
`;

exports[`Sample event with all fields defined should match snapshot 1`] = `
<tr
  className="activity-run"
>
  <td
    className="data-date"
  >
    <span
      className="value"
    >
      Today
    </span>
  </td>
  <td
    className="data-category"
  >
    <span
      className="glyphicon glyphicon-road"
    />
     
    <span
      className="value"
    >
      casual
    </span>
  </td>
  <td
    className="data-distance"
  >
    <span
      className="value"
    >
      5 mi
    </span>
  </td>
  <td
    className="data-duration"
  >
    <span
      className="value"
    >
      50m 0s
    </span>
  </td>
  <td
    className="data-notes"
  >
    <span
      className="value"
    >
      Just a test.
    </span>
  </td>
  <td
    className="data-actions"
  >
    <a
      className="glyphicon glyphicon-heart-empty"
      href="#"
    />
    <a
      className="glyphicon glyphicon-pencil"
      href="#"
    />
    <a
      className="glyphicon glyphicon glyphicon-trash"
      href="#"
    />
  </td>
</tr>
`;
